lib/RemoteAST/RemoteAST.cpp

```diff
@@ -342,6 +342,11 @@ class RemoteASTTypeBuilder {
         if (inOutArgs[i]) arg = InOutType::get(arg);
         elts.push_back(arg);
       }
+
+      if (elts.empty())
+        input = TupleType::getEmpty(Ctx);
+      else
+        input = TupleType::get(elts, Ctx);
```

Comment:
You don't need this "if" here; `TupleType::get(elts, Ctx)` works with zero elements just as well.

Reply:
Ah, should have looked at what `getEmpty(...)` does, I though the fact we have it might produce something more optimized from memory point of view :)

Reply:
It'll hit hash table and return the same thing as `TupleType::getEmpty(Ctx)`.  If there's anything to optimize here, it's an `empty()` check in `TupleType::get()` itself.

Reply:
It doesn't really matter either way.